charts/redpanda/statefulset.go

```diff
@@ -45,6 +45,9 @@ const (
 	// https://github.com/kubernetes/kubernetes/blob/c6669ea7d61af98da3a2aa8c1d2cdc9c2c57080a/plugin/pkg/admission/serviceaccount/admission.go#L55-L57
 	//nolint:gosec
 	DefaultAPITokenMountPath = "/var/run/secrets/kubernetes.io/serviceaccount"
+
+	NodePoolLabelName      = "cluster.redpanda.com/nodepool-name"
+	NodePoolLabelNamespace = "cluster.redpanda.com/nodepool-namespace"
```

Comment:
Our clusterref type doesn't have a namespace on it so we can safely assume all references are intra-namespace. Do we really need this label?

I could also see an argument for setting the ownerref of nodepools to the cluster as the "binding" process.

Reply:
Yeah, I think I can drop the label. As far as ownership is concerned, not sure that would play well with things like cascading deletion. In general I’d prefer if we not try and take ownership of user-controlled resources